ci-operator/step-registry/cucushift/installer/check/azure/confidential/trustedlaunch/cucushift-installer-check-azure-confidential-trustedlaunch-commands.sh

```diff
@@ -0,0 +1,167 @@
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+critical_check_result=0
+
+# master machine check
+master_encryptionathost=$(yq-go r "${INSTALL_CONFIG}" "controlPlane.platform.azure.encryptionAtHost")
+master_security_type=$(yq-go r "${INSTALL_CONFIG}" "controlPlane.platform.azure.settings.securityType")
+master_secure_boot=$(yq-go r "${INSTALL_CONFIG}" "controlPlane.platform.azure.settings.trustedLaunch.uefiSettings.secureBoot")
+master_vtpm=$(yq-go r "${INSTALL_CONFIG}" "controlPlane.platform.azure.settings.trustedLaunch.uefiSettings.virtualizedTrustedPlatformModule")
+
+master_nodes=$(oc get machine -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers | awk '{print $1}')
+for master in ${master_nodes}; do
+    echo "--- check master node ${master} ---"
+
+    security_profile_output=$(mktemp)
+    az vm show -n "${master}" -g "${RESOURCE_GROUP}" -ojson | jq -r '.securityProfile' 1>"${security_profile_output}"
+
+    echo "node ${master} security profile"
+    cat "${security_profile_output}"
+    if [[ $(< "${security_profile_output}") == "null" ]]; then
+        echo "node ${master} security profile is null, check failed!"
+        critical_check_result=1
+        continue
+    fi
+
+    if property_check "${master}" "encryptionAtHost" "${master_encryptionathost}" "${security_profile_output}" ; then
+        echo "property encryptionAtHost check passed."
+    else
+        echo "property encryptionAtHost check failed."
+        critical_check_result=1
+    fi
+
+    if property_check "${master}" "securityType" "${master_security_type}" "${security_profile_output}" ; then
+        echo "property securityType check passed."
+    else
+        echo "property securityType check failed."
+        critical_check_result=1
+    fi
+
+    if property_check "${master}" "uefiSettings.secureBootEnabled" "${master_secure_boot}" "${security_profile_output}"; then
+        echo "property secureBootEnabled check passed."
+    else
+        echo "property secureBootEnabled check failed."
+        critical_check_result=1
+    fi
+
+    if property_check "${master}" "uefiSettings.vTpmEnabled" "${master_vtpm}" "${security_profile_output}"; then
+        echo "property vTpmEnabled check passed."
+    else
+        echo "property vTpmEnabled check failed."
+        critical_check_result=1
+    fi
+    rm -f "${security_profile_output}"
+done
+
+# worker machines check
+worker_encryptionathost=$(yq-go r "${INSTALL_CONFIG}" "compute[0].platform.azure.encryptionAtHost")
+worker_security_type=$(yq-go r "${INSTALL_CONFIG}" "compute[0].platform.azure.settings.securityType")
+worker_secure_boot=$(yq-go r "${INSTALL_CONFIG}" "compute[0].platform.azure.settings.trustedLaunch.uefiSettings.secureBoot")
+worker_vtpm=$(yq-go r "${INSTALL_CONFIG}" "compute[0].platform.azure.settings.trustedLaunch.uefiSettings.virtualizedTrustedPlatformModule")
```

Comment:
Because we are setting instal-config.yaml in a static way once enabled `trustedlaunch`, I suggest to set the hard code here, too, instead of reading install-config.yaml.
Another point, because the install-config.yaml is patched again and again, we can use hard code to avoid some other step bugs impact.